components/zoho_desk/zoho_desk.app.mjs

```diff
@@ -1,11 +1,354 @@
+      });
+    },
+    searchContacts(args = {}) {
+      return this.makeRequest({
+        path: "/contacts/search",
+        ...args,
+      });
+    },
+    getDepartments(args = {}) {
+      return this.makeRequest({
+        path: "/departments",
+        ...args,
+      });
+    },
+    createTicket(args = {}) {
+      return this.makeRequest({
+        method: "post",
+        path: "/tickets",
+        ...args,
+      });
+    },
+    updateTicket({
+      ticketId, ...args
+    } = {}) {
+      return this.makeRequest({
+        method: "patch",
+        path: `/tickets/${ticketId}`,
+        ...args,
+      });
+    },
+    createTicketComment({
+      ticketId, ...args
+    } = {}) {
+      return this.makeRequest({
+        method: "post",
+        path: `/tickets/${ticketId}/comments`,
+        ...args,
+      });
+    },
+    getTickets(args = {}) {
+      return this.makeRequest({
+        path: "/tickets",
+        ...args,
+      });
+    },
+    searchTickets(args = {}) {
+      return this.makeRequest({
+        path: "/tickets/search",
+        ...args,
+      });
+    },
+    createTicketAttachment({
+      ticketId, ...args
+    } = {}) {
+      return this.makeRequest({
+        method: "post",
+        path: `/tickets/${ticketId}/attachments`,
+        ...args,
+      });
+    },
+    getTicketAttachments({
+      ticketId, ...args
+    } = {}) {
+      return this.makeRequest({
+        path: `/tickets/${ticketId}/attachments`,
+        ...args,
+      });
+    },
+    getTicketComments({
+      ticketId, ...args
+    } = {}) {
+      return this.makeRequest({
+        path: `/tickets/${ticketId}/comments`,
+        ...args,
+      });
+    },
+    sendReply({
+      ticketId, ...args
+    } = {}) {
+      return this.makeRequest({
+        method: "post",
+        path: `/tickets/${ticketId}/sendReply`,
+        ...args,
+      });
+    },
+    getContacts(args = {}) {
+      return this.makeRequest({
+        path: "/contacts",
+        ...args,
+      });
+    },
+    getSupportEmailAddresses(args = {}) {
+      return this.makeRequest({
+        path: "/supportEmailAddress",
+        ...args,
+      });
+    },
+    getAccounts(args = {}) {
+      return this.makeRequest({
+        path: "/accounts",
+        ...args,
+      });
+    },
+    getAgents(args = {}) {
+      return this.makeRequest({
+        path: "/agents",
+        ...args,
+      });
+    },
+    async getResourcesOptions({
+      orgId, departmentId, prevContext, resourceFn, resourceMapper,
+    }) {
+      const { from = 1 } = prevContext;
+      if (from === null) {
+        return [];
+      }
+      const { data: resources } =
+        await resourceFn({
+          headers: {
+            orgId,
+          },
+          params: {
+            departmentId,
+            from,
+            limit: constants.DEFAULT_LIMIT,
+          },
+        });
+      const currentLen = resources.length;
+      const options = resources.map(resourceMapper);
```

Comment:
The prop pagination works, but it gets errors when it reaches the bottom of the list. Adding `?` fixed it for me.

<img width="892" alt="Screen Shot 2022-09-16 at 4 27 27 PM" src="https://user-images.githubusercontent.com/10889229/190727599-736a3277-53d4-46fb-9d9e-8fd9fed18d94.png">

<img width="888" alt="Screen Shot 2022-09-16 at 4 25 44 PM" src="https://user-images.githubusercontent.com/10889229/190727657-65eb5089-66ba-4b17-996c-8441b96b33e0.png">


```suggestion
      const currentLen = resources?.length;
      const options = resources?.map(resourceMapper);
```